api/v1/configurationpolicy_types.go

```diff
@@ -149,62 +121,89 @@ func (e EvaluationInterval) parseInterval(interval string) (time.Duration, error
+	MustHave ComplianceType = "MustHave"
+
+	// MustOnlyHave is a ComplianceType to match an object definition exactly with the object.
+	MustOnlyHave ComplianceType = "MustOnlyHave"
+)
+
+func (c ComplianceType) IsMustHave() bool {
+	return strings.EqualFold(string(c), string(MustHave))
 }
 
-// ObjectTemplate describes how an object should look
-type ObjectTemplate struct {
-	// ComplianceType specifies whether it is: musthave, mustnothave, mustonlyhave
-	ComplianceType ComplianceType `json:"complianceType"`
+func (c ComplianceType) IsMustOnlyHave() bool {
+	return strings.EqualFold(string(c), string(MustOnlyHave))
+}
+
+func (c ComplianceType) IsMustNotHave() bool {
+	return strings.EqualFold(string(c), string(MustNotHave))
+}
+
+// MetadataComplianceType describes how the labels and annotations of objects on the cluster should
+// be compared with the object definition of the configuration policy. The supported options are
+// `MustHave` or `MustOnlyHave`. The default value is the value defined in `complianceType` for the
+// object template.
+//
+// +kubebuilder:validation:Enum=MustHave;Musthave;musthave;MustOnlyHave;Mustonlyhave;mustonlyhave
+type MetadataComplianceType string
+
+// +kubebuilder:validation:Enum=Log;InStatus;None
+type RecordDiff string
+
+const (
+	RecordDiffLog      RecordDiff = "Log"
+	RecordDiffInStatus RecordDiff = "InStatus"
+	RecordDiffNone     RecordDiff = "None"
+	// Censored is only used as an internal value to indicate a diff shouldn't be automatically generated.
+	RecordDiffCensored RecordDiff = "Censored"
+)
 
+// ObjectTemplate describes the desired state of an object on the cluster.
+type ObjectTemplate struct {
+	ComplianceType         ComplianceType         `json:"complianceType"`
 	MetadataComplianceType MetadataComplianceType `json:"metadataComplianceType,omitempty"`
 
-	// ObjectDefinition defines required fields for the object
+	// ObjectDefinition defines required fields to be compared with objects on the cluster.
+	//
 	// +kubebuilder:pruning:PreserveUnknownFields
 	ObjectDefinition runtime.RawExtension `json:"objectDefinition"`
 
-	// RecordDiff specifies whether (and where) to log the diff between the object on the
-	// cluster and the objectDefinition in the policy. Defaults to "None" when the object kind is
-	// ConfigMap, OAuthAccessToken, OAuthAuthorizeTokens, Route, or Secret. Defaults to "InStatus" otherwise.
+	// RecordDiff specifies whether and where to log the difference between the object on the cluster
+	// and the `objectDefinition` in the policy. The supported options are `InStatus` to record the
+	// difference in the policy status field, `Log` to log the difference in the
+	// config-policy-controller pod, and `None` to not log the diff. The default value is `None` for
+	// object kinds that include sensitive data including `ConfigMap`, `OAuthAccessToken`,
+	// `OAuthAuthorizeTokens`, `Route`, and `Secret`. For all other kinds, the default value is
+	// `InStatus`.
```

Comment:
FYI @mprahl I updated the description for `recordDiff`.

Reply:
Updated here:
https://github.com/open-cluster-management-io/config-policy-controller/pull/248

Thanks for the suggestion!